### ISSUE TITLE
[Bug]: Top navigation icon color not consistent and black on black

### ISSUE THREAD

**Issue by violoncelloCH** [nextcloud/server#43452]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

Since the update to 28.0.2, my top navigation bar looks like this:

![Screenshot from 2024-02-08 13-11-27](https://github.com/nextcloud/server/assets/22591354/aa4c3d3e-70c0-4610-813f-90169e5c760a)
(Note the icons on the top right as well as the welcome text on the dashbord being black on black background.)

This happens with theming color of #F06000 (or any other "ligth" color) together with a background image which is dark (at the top).

### Steps to reproduce

1. choose a light theming color
2. choose a background image which is dark (at the top)
3. see search, addressbook and assistant icons being black on dark background while the App icons are still white

### Expected behavior

The icons and text color should adapt to the background image like the App icons at the top left.

### Installation method

None

### Nextcloud Server version

28

### Operating system

Debian/Ubuntu

### PHP engine version

None

### Web server

None

### Database engine version

None

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
-
```

### List of activated Apps

```shell
-
```

### Nextcloud Signing status

```shell
-
```

**Comment by violoncelloCH** [comment]
ahh, sorry @provokateurin, I misunderstood @jancborchardt's message then... anyone else to who knows how this is implemented?

**Comment by jancborchardt** [comment]
@skjnldsv maybe or @juliushaertl regarding theming?

**Comment by susnux** [comment]
Duplicate and already fixed, will be fixed with 28.0.3

Ref: https://github.com/nextcloud/server/issues/43298
